leetcode_bwang

```diff
@@ -0,0 +1,28 @@
+class Solution {
+    public List<List<Integer>> permuteUnique(int[] nums) {
+        List<List<Integer>> wraplist = new ArrayList<>();
+        List<Integer> sublist = new ArrayList<>();
+        boolean[] visited = new boolean[nums.length];
+        Arrays.sort(nums);
+        dfs(wraplist, sublist, nums, visited);
+        return wraplist;
+    }
+    
+    public void dfs(List<List<Integer>> wraplist, List<Integer> sublist, int[] nums, boolean[] visited) {
+        if (sublist.size() == nums.length) {
+            wraplist.add(new ArrayList<>(sublist));
+            return;
+        }
+        
```

Comment:
For recursive calls, you wanna consider the base case first. Under which condition do you want this recursive call to end and return? In our case, it's when you put all of numbers in a list container, so the size of list needs to equals total count of numbers.